src/extension.ts

```diff
@@ -162,16 +170,40 @@ export async function activate(extensionContext: vscode.ExtensionContext): Promi
   const configurationApi = await k8s.extension.configuration.v1_1;
   if (configurationApi.available) {
     const confApi = configurationApi.api;
+    setKubeConfigPath(confApi);
     confApi.onDidChangeContext(async () => {
       // eslint-disable-next-line no-use-before-define
       await checkClusterVersion();
+      setKubeConfigPath(confApi);
+    });
+    confApi.onDidChangeNamespace(() => {
+      setKubeConfigPath(confApi);
+    });
+    confApi.onDidChangeKubeconfigPath(() => {
+      setKubeConfigPath(confApi);
     });
   }
 
   // extensionContext.subscriptions.push(disposable);
   disposable.forEach((value) => extensionContext.subscriptions.push(value));
 }
 
+function setKubeConfigPath(confApi: ConfigurationV1_1): void {
+  const configPath = confApi.getKubeconfigPath();
+  if (configPath?.['hostPath']) {
+    process.env.KUBECONFIG = configPath?.['hostPath'];
+    functionExplorer.refresh();
```

Comment:
This could be optimized. There is some duplicated code in the two `if`

Reply:
done thanks :)